module/VuFindConsole/src/VuFindConsole/Command/ScheduledSearch/NotifyCommand.php

```diff
@@ -294,35 +221,24 @@ protected function validateSchedule($todayTime, $lastTime, $s)
     }
 
     /**
-     * Load and validate a user object associated with the search; return false
+     * Load and validate a user object associated with the search; return null
      * if there is a problem.
      *
-     * @param \VuFind\Db\Row\Search $s Current search row.
+     * @param SearchEntityInterface $s Current search row.
      *
-     * @return UserEntityInterface|bool
+     * @return ?UserEntityInterface
      */
     protected function getUserForSearch($s)
     {
-        // Use a static variable to hold the last accessed user (to spare duplicate
-        // database lookups, since we're loading rows in user order).
-        static $user = false;
-
-        if ($user === false || $s->user_id != $user->getId()) {
-            if (!$user = $this->userTable->getById($s->user_id)) {
-                $user = false;  // make sure static variable is cleared
-                $this->warn(
-                    'Search ' . $s->id . ': user ' . $s->user_id
-                    . ' does not exist '
-                );
-                return false;
-            }
+        if (!$user = $s->getUser()) {
```

Comment:
Due to changes in the way the search and user entities relate to one another, it seemed cleanest to eliminate the user caching logic in order to simplify the code and reduce dependencies. In some situations, this may put slightly higher load on the database, but looking up a user by primary key is not an expensive operation, so I'm not too concerned about it.